app-backend/database/src/main/scala/com/azavea/rf/database/tables/Projects.scala

```diff
@@ -329,13 +378,44 @@ object Projects extends TableQuery(tag => new Projects(tag)) with LazyLogging {
     */
   def deleteScenesFromProject(sceneIds: Seq[UUID], projectId: UUID)
                             (implicit database: DB): Future[Int] = {
-
-    val sceneProjectJoinQuery = for {
-      s <- ScenesToProjects if s.sceneId.inSet(sceneIds) && s.projectId === projectId
-    } yield s
+    val remainingScenesQuery = Scenes
+      .filter { scene =>
+        scene.id in ScenesToProjects.filter(_.projectId === projectId).filterNot(_.sceneId inSet sceneIds).map(_.sceneId)
+      }
 
     database.db.run {
-      sceneProjectJoinQuery.delete
+      remainingScenesQuery.result
+    } flatMap { remainingScenes =>
+      val newProjectExtentGeometry = remainingScenes.length match {
+        case 0 => None
+        case _ => {
+          val footprintEnvelopes = remainingScenes flatMap { _.dataFootprint map { _.geom.envelope } }
```

Comment:
Since you do `footprints --> combined footprints` more than once, there should be some `def f(footprints: Seq[Option[Projected[Geometry]]]): Option[Projected[Geometry]] = ???` that handles that reduction